analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -168,133 +150,254 @@ public static void LoopWithBreak(System.Collections.Generic.IEnumerable<string>
+                Console.WriteLine(foo.Value.ToString());    // FN
             }
             if (foo == null)
             {
-                Console.WriteLine(foo.Value.ToString());    // Compliant, unreachable
+                Console.WriteLine(foo.Value.ToString());    // FN
             }
             if (foo != null)
             {
                 Console.WriteLine(foo.Value.ToString());
             }
         }
-        else
+    }
+
+    public void NestedIsolated1(DateTime? value)
+    {
+        if (foo == value)   // Relationship is added here
         {
-            if (foo.HasValue)
+            if (value.HasValue)
             {
-                Console.WriteLine(foo.Value.ToString());    // Compliant, unreachable
+                if (!foo.HasValue)
+                {
+                    Console.WriteLine(foo.Value.ToString());    // Compliant
+                }
             }
-            if (!foo.HasValue)
+        }
+    }
+
+    public void NestedIsolated2(DateTime? value)
+    {
+        if (foo == value)   // Relationship is added here
+        {
+            if (!value.HasValue)
             {
-                Console.WriteLine(foo.Value.ToString());    // FIXME Non-compliant
+                if (foo == null)
+                {
+                    Console.WriteLine(foo.Value.ToString());    // FN
+                }
             }
-            if (foo == null)
+        }
+    }
+
+    public void NestedCombined(DateTime? value)
+    {
+        if (foo == value)   // Relationship is added here
+        {
+            if (value.HasValue)
             {
-                Console.WriteLine(foo.Value.ToString());    // FN. It works as expected when isolated, see NestedIsolated2() above
+                if (foo.HasValue)
+                {
+                    Console.WriteLine(foo.Value.ToString());
+                }
+                if (!foo.HasValue)
+                {
+                    Console.WriteLine(foo.Value.ToString());
+                }
+                if (foo == null)
+                {
+                    // FIXME the following
+                    Console.WriteLine(foo.Value.ToString());    // FN
+                }
+                if (foo != null)
+                {
+                    Console.WriteLine(foo.Value.ToString());
+                }
             }
-            if (foo != null)
+            else
             {
-                Console.WriteLine(foo.Value.ToString());    // Compliant, unreachable
+                if (foo.HasValue)
+                {
+                    Console.WriteLine(foo.Value.ToString());    // Compliant, unreachable
+                }
+                if (!foo.HasValue)
+                {
+                    Console.WriteLine(foo.Value.ToString());    // FIXME Non-compliant
+                }
+                if (foo == null)
+                {
+                    Console.WriteLine(foo.Value.ToString());    // FN
+                }
```

Comment:
These are basically the same cases. Why do they not have the same comments?

Reply:
Indeed. I have aligned them, keeping into account that `value.HasValue` make them somewhat antisimmetric:
- compliance due to non-emptiness becomes compliance due to unreachability
- compliance due to unreachability becomes non-compliance due to emptiness